main/src/com/google/refine/commands/GetAllPreferencesCommand.java

```diff
@@ -46,13 +46,18 @@
 import com.google.refine.preference.PreferenceStore;
 
 public class GetAllPreferencesCommand extends Command {
-	/**
-	 * The command uses POST (not sure why?) but does not actually modify any state
-	 * so it does not require CSRF.
-	 */
```

Comment:
@tfmorris: I'm not sure I would change that. I bet that the programer's concern at the time with a GET request is that a non-authorized user could have access to private preference's data.

That would be the purpose of the CSRF. Prevent intruders access.

Regards, A.

Reply:
That comment is, I believe, from @wetneb saying that he's not going to add CSRF protection to that post, so it's not currently protected. Also `get-preference` is already available via GET. This makes the API more regular, in my opinion, but I'm open to other opinions. @wetneb ?

Reply:
@tfmorris: this is documented here. I was wrong, CSRF are not needed for this POST, so it's not a security thing.

https://github.com/OpenRefine/OpenRefine/wiki/Changes-for-3.3

```
The following commands can still be used with POST but do not require CSRF tokens (although supplying one will not fail), as they do not incur a state change in the backend:
- compute-clusters;
- compute-facets;
- export-project;
- export-rows;
- get-all-preferences;
- get-importing-configuration;
- get-importing-job-status;
- get-rows;
- load-language.
```

Regards, Antoine

Reply:
Yes it's a good idea to deprecate this - however I am worried that just adding a Java annotation will not be very visible to any client that calls this method using POST. We should make the deprecation more visible, for instance by a warning-level log line, or an additional message added to the JSON response. Perhaps it would be worth keeping the comment (or adapt it) to explain why we are deprecating this.